sdk/Trace/TracerProvider.php

```diff
@@ -27,10 +28,17 @@ final class TracerProvider implements API\TracerProvider
      */
     private $resource;
 
-    public function __construct(?ResourceInfo $resource = null)
+    /**
+     * @var Sampler
+     */
+    private $sampler;
+
+    public function __construct(?ResourceInfo $resource = null, ?Sampler $sampler = null)
     {
         $this->spanProcessors = new SpanMultiProcessor();
         $this->resource = $resource ?? ResourceInfo::emptyResource();
+        $this->sampler = $sampler ?? new AlwaysOnSampler();
```

Comment:
Should this be `AlwaysOnSampler()` if no sampler is provided?  Maybe it should be `AlwaysOffSampler()`?  Don't have strong opinions here.

Reply:
It could go either way.  My thinking with this was ease of use.  It's easier to use if spans get recorded by default vs a user having to figure out a sampler to add just to get spans recorded.  I thought more users would prefer to have spans recorded vs not recorded.

What do you think?

Reply:
Sounds good to me.  Easy enough to change later on if need be.